test/extended/default_forcePull_test.go

```diff
@@ -0,0 +1,234 @@
+// +build default
+
+package extended
+
+import (
+	"fmt"
+	"path/filepath"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	"time"
+)
+
+var (
+	fpTrueS2I   = filepath.Join("fixtures", "forcepull-true-s2i.json")
+	fpFalseS2I  = filepath.Join("fixtures", "forcepull-false-s2i.json")
+	fpTrueDock  = filepath.Join("fixtures", "forcepull-true-dock.json")
+	fpFalseDock = filepath.Join("fixtures", "forcepull-false-dock.json")
+	fpTrueCust  = filepath.Join("fixtures", "forcepull-true-cust.json")
+	fpFalseCust = filepath.Join("fixtures", "forcepull-false-cust.json")
+	s2iDockBldr = "docker.io/openshift/ruby-20-centos7"
+	custBldr    = "docker.io/openshift/origin-custom-docker-builder"
+)
+
+/*
+If docker.io is not responding to requests in a timely manner, this test suite will be adversely affected.
+
+If you suspect such a situation, attempt pulling some openshift images other than ruby-20-centos7 or origin-custom-docker-builder
+while this test is running and compare results.  Restarting your docker daemon, assuming you can ping docker.io quickly, could
+be a quick fix.
+*/
+
+func corruptImage(corruptee, corruptor, strategy string) {
+	By(fmt.Sprintf("\n%s****BEFORE EACH FORCE PULL*****:  Calling docker tag to corrupt %s builder image %s by tagging %s", time.Now().Format(time.RFC850), strategy, corruptee, corruptor))
+
+	cerr := exutil.CorruptImage(corruptee, corruptor)
+
+	By(fmt.Sprintf("\n%s****BEFORE EACH FORCE PULL*****:  Tagging %s to %s complete with err %v", time.Now().Format(time.RFC850), corruptor, corruptee, cerr))
+	Expect(cerr).NotTo(HaveOccurred())
+}
+
+func resetImage() {
+	// reset both images just to be sure
+	By(fmt.Sprintf("\n%s****FORCE PULL*****:  Calling docker pull to reset images", time.Now().Format(time.RFC850)))
+
+	serr := exutil.ResetImage(s2iDockBldr)
+
+	By(fmt.Sprintf("\n%s****FORCE PULL*****:  Reset pull for %s complete with err %v", time.Now().Format(time.RFC850), s2iDockBldr, serr))
+	Expect(serr).NotTo(HaveOccurred())
+
+	cerr := exutil.ResetImage(custBldr)
+
+	By(fmt.Sprintf("\n%s****FORCE PULL*****:  Reset pull for %s complete with err %v", time.Now().Format(time.RFC850), custBldr, cerr))
+	Expect(cerr).NotTo(HaveOccurred())
+
+	// a small sleep between a potential "AfterEach" and the next potential "BeforeEach", just to make sure the docker daemon has processed our reset of images
+	time.Sleep(1 * time.Second)
```

Comment:
The comment above pertains more to the sleep below.  It is possible I was just paranoid and could just lose the sleep / don't need to verify the pull.